src/main/java/mekanism/client/render/transmitter/RenderLogisticalTransporter.java

```diff
@@ -53,81 +62,118 @@ public void render(TileEntityLogisticalTransporter transporter, double x, double
-
-                if (obj.equals(new Coord4D(transporter.getPos(), transporter.getWorld())) && pos.sideHit != null) {
+            if (!pushed) {
+                GlStateManager.pushMatrix();
+                pushed = true;
+            }
+            ItemStack itemStack = mc.player.inventory.getCurrentItem();
+            if (!itemStack.isEmpty() && itemStack.getItem() instanceof ItemConfigurator) {
+                RayTraceResult pos = mc.player.rayTrace(8.0D, 1.0F);
+                if (pos != null && pos.sideHit != null && pos.getBlockPos().equals(transporter.getPos())) {
                     int mode = ((TileEntityDiversionTransporter) transporter).modes[pos.sideHit.ordinal()];
-
-                    pushTransporter();
-
-                    GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.8F);
-
-                    bindTexture(MekanismRenderer.getBlocksTexture());
-                    GL11.glTranslatef((float) x, (float) y, (float) z);
-                    GL11.glScalef(0.5F, 0.5F, 0.5F);
-                    GL11.glTranslatef(0.5F, 0.5F, 0.5F);
-
-                    int display = getOverlayDisplay(world, pos.sideHit, mode).display;
-                    GL11.glCallList(display);
-
-                    popTransporter();
+                    GlStateManager.pushMatrix();
+                    GlStateManager.enableCull();
+                    GlStateManager.disableLighting();
+                    GlowInfo glowInfo = GLSMHelper.enableGlow();
+                    GlStateManager.shadeModel(GL11.GL_SMOOTH);
+                    GlStateManager.disableAlpha();
+                    GlStateManager.enableBlend();
+                    GlStateManager.blendFunc(SourceFactor.SRC_ALPHA, DestFactor.ONE_MINUS_SRC_ALPHA);
+                    GlStateManager.color(1, 1, 1, 0.8F);
+                    bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    GlStateManager.translate((float) x, (float) y, (float) z);
+                    GlStateManager.scale(0.5F, 0.5F, 0.5F);
+                    GlStateManager.translate(0.5F, 0.5F, 0.5F);
+
+                    int display = getOverlayDisplay(pos.sideHit, mode).display;
+                    GlStateManager.callList(display);
+
+                    GLSMHelper.resetColor();
+                    GlStateManager.disableBlend();
+                    GlStateManager.enableAlpha();
+                    GLSMHelper.disableGlow(glowInfo);
+                    GlStateManager.enableLighting();
+                    GlStateManager.disableCull();
+                    GlStateManager.popMatrix();
                 }
             }
         }
+        if (pushed) {
+            //If we did anything we need to pop the matrix we pushed
+            GlStateManager.popMatrix();
+        }
+    }
 
-        GL11.glPopMatrix();
+    /**
+     * Shrink the in transit list as much as possible. Don't try to render things that are in the same spot with the same color
```

Comment:
Good catch, I ended up going through and changing the implementation to be more accurate and using a custom object in a single set rather than having multiple sets. This also let me add the checks for ItemStack type which I was unsure if was needed before. (Given I thought about it some more and realized that is in fact the *most* important check for deduplicating what needs to be rendered that there is).